src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2714,26 +2714,29 @@ private bool TryVisitConditionalAccess(BoundExpression node, [NotNullWhen(true)]
             stateWhenNotNull = default;
             return false;
 
-            // "State when not null" can only propagate out of a conditional access if
-            // it is not subject to a user-defined conversion whose parameter is not of a non-nullable value type.
-            static bool isAcceptableConversion(BoundConditionalAccess operand, Conversion conversion)
-            {
-                if (conversion.Kind is not (ConversionKind.ImplicitUserDefined or ConversionKind.ExplicitUserDefined))
-                {
-                    return true;
-                }
+        }
 
-                var method = conversion.Method;
-                Debug.Assert(method is not null);
-                Debug.Assert(method.ParameterCount is 1);
-                var param = method.Parameters[0];
-                if (operand.Type.IsNullableType() && param.Type.IsNonNullableValueType())
-                {
-                    return true;
-                }
+        /// <summary>
+        /// "State when not null" can only propagate out of a conditional access if
+        /// it is not subject to a user-defined conversion whose parameter is not of a non-nullable value type.
+        /// </summary>
+        protected static bool CanPropagateStateWhenNotNull(BoundConditionalAccess operand, Conversion conversion)
+        {
+            if (conversion.Kind is not (ConversionKind.ImplicitUserDefined or ConversionKind.ExplicitUserDefined))
+            {
+                return true;
+            }
 
-                return false;
+            var method = conversion.Method;
+            Debug.Assert(method is not null);
+            Debug.Assert(method.ParameterCount is 1);
+            var param = method.Parameters[0];
+            if (operand.Type.IsNullableType() && param.Type.IsNonNullableValueType())
```

Comment:
>operand.Type.IsNullableType() && param.Type.IsNonNullableValueType() [](http://example.com/codeflow?start=16&length=68)

Consider just returing this condition.

Reply:
I will make this adjustment in a future PR for this feature.